### PR TITLE
[hotfix][streaming] rename local variable name and fix some typos

### PR DESCRIPTION
**Pull request by JingGe**
## What is the purpose of the change

Fix some typos while reading the mailbox code. 

## Brief change log

  - fix the wrong local variable name in MailboxProcessor
  - fix some typos in java doc

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)

### CODE REVIEW

**Comment by JingGe** [comment]
@MartijnVisser Would you like to review this PR? Many thanks!

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java [hunk 1]
```diff
@@ -274,7 +274,7 @@
 
     protected final MailboxProcessor mailboxProcessor;
 
-    final MailboxExecutor mainMailboxExecutor;
+    protected final MailboxExecutor mainMailboxExecutor;
```
Comment (MartijnVisser) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java:277]: I can't verify if this is indeed correct, perhaps @fapaul of @AHeise can verify this.

**Review by MartijnVisser — COMMENTED** [review]
Thanks for the fixes @JingGe - I can't verify the instance variable fix, but other then that it looks OK.

**Comment by JingGe** [comment]
@flinkbot run azure

**Comment by MartijnVisser** [comment]
@flinkbot run azure

**Comment by fapaul** [comment]
@pnowojski I guess the visibility change in the `StreamTask` is not a problem but can you double check that this was an oversight?

**Comment by pnowojski** [comment]
I don't mind too much one way or another. I think it was package-private as it's more narrow scope compared to `protected`, and the smaller the scope, usually the better. But IMO it doesn't matter here too much.

**Comment by JingGe** [comment]
> I don't mind too much one way or another. I think it was package-private as it's more narrow scope compared to `protected`, and the smaller the scope, usually the better. But IMO it doesn't matter here too much.

Yeah, I had the same thought. But after I checked the code, it is defined as package-private but used as protected. In this case, it even enlarge the visibility scope, because classes within this package but do not inherit `StreamTask` are able to access it. I guess it is not intended to work as package-private and is against OO design and might cause potential issue. Another issue is that any subclass of `StreamTask` that is not in the same package will not be able to access this instance variable. If we want to limit all subclasses of `StreamTask` be in the same package, we should control it via ArchUnit rule.

**Comment by JingGe** [comment]
> > Yeah, I had the same thought. But after I checked the code, it is defined as package-private but used as protected. In this case, it even enlarge the visibility scope, because classes within this package but do not inherit `StreamTask` are able to access it.
> 
> In Java, `protected` implies `package-private`, so it's strictly larger scope.

I see, thanks. It then makes sense to keep package-private which should be only changed if any subclass need to be in a different package. I will remove this change.

**Review by pnowojski — APPROVED** [review]
Thanks for the update. LGTM

**Pull request merged**